### PR TITLE
Standardize edit-in-place and hyperlink UIs

### PR DESCRIPTION
**Pull request by chiaberry**
## Associated issues

https://github.com/cityofaustin/atd-data-tech/issues/7870

## Testing
**URL to test:** 
https://7870-edit-in-place--atd-moped-main.netlify.app/moped/dashboard
https://7870-edit-in-place--atd-moped-main.netlify.app/moped/projects/323/
**Steps to test:**

Check in the summary view that you don't see any pencils. That clicking a field will open the edit capability. 

On the signals dashboard you shouldn't be able to edit a row. Empty editable fields will have a hyphen as a placeholder. 
---
#### Ship list
- [x] Code reviewed 
- [x] Product manager approved
- [x] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#gid=776973707)

### CODE REVIEW

**Comment by sergiogcx** [comment]
@chiaberry Awesome job. There needs to be a discussion with @amenity about the urls, I can't remember if it was discussed and what the conclusion was about those links:

File: moped-editor/src/views/projects/signalProjectTable/SignalProjectTable.js [hunk 1]
```diff
@@ -447,7 +394,7 @@ const SignalProjectTable = () => {
         updatedProjectObject[columnDef.field] = newData;
 
         updatedProjectObject["entity_id"] =
-          updatedProjectObject.project_sponsor.entity_id;
+          updatedProjectObject?.project_sponsor?.entity_id;
```
Comment (sergiogcx) [inline moped-editor/src/views/projects/signalProjectTable/SignalProjectTable.js:397]: Nice 👍

**Review by sergiogcx — APPROVED** [review]

**Review by sergiogcx — CHANGES_REQUESTED** [review]
Well, on second thought there might be regressions if we don't have a discussion. Changing to request changes for now.

**Review by sergiogcx — APPROVED** [review]

**Review by amenity — APPROVED** [review]
Fabulous, thanks! 🙏

**Pull request merged**